test/test_core/test_tree_policies/test_UCT_TreePolicy.cpp

```diff
@@ -0,0 +1,111 @@
+#include <UCT_TreePolicy.h>
+#include <gtest/gtest.h>
+#include <memory>
+#include <string>
+
+State tempState(std::nullopt);
+
+class UCT_TreePolicyTest : public ::testing::Test {
+
+  protected:
+    void SetUp() override {
+
+        // root
+        State rootNode = State("rootNode");
+        root = SearchNode::create_SearchNode(nullptr, rootNode, false);
+
+        // root as parent
+        State rootChild1State = State("rootChild1");
+        State rootChild2State = State("rootChild2");
+        std::shared_ptr<SearchNode> rootChild1 = SearchNode::create_SearchNode(root, rootChild1State, false);
+        std::shared_ptr<SearchNode> rootChild2 = SearchNode::create_SearchNode(root, rootChild2State, false);
+
+        std::function<std::shared_ptr<SearchNode>(std::shared_ptr<SearchNode>)> sExp =
+            std::bind(&UCT_TreePolicyTest::simpleExpand, this, std::placeholders::_1);
+        std::function<std::shared_ptr<SearchNode>(std::shared_ptr<SearchNode>)> sChild =
+            std::bind(&UCT_TreePolicyTest::simpleChild, this, std::placeholders::_1);
+        treePolicy = UCT_TreePolicy(sExp, sChild);
+    }
+
+    std::shared_ptr<SearchNode> root = SearchNode::create_SearchNode(nullptr, tempState, false);
+    std::function<std::shared_ptr<SearchNode>(std::shared_ptr<SearchNode> node)> placeholderFunc = [](std::shared_ptr<SearchNode> node) {
+        return node;
+    };
+    UCT_TreePolicy treePolicy = UCT_TreePolicy(placeholderFunc, placeholderFunc);
+
+    std::shared_ptr<SearchNode> simpleExpand(std::shared_ptr<SearchNode> node) {
+        State state = node->unvisited_child_states.at(0);
+        std::shared_ptr<SearchNode> expanded_node = SearchNode::create_SearchNode(node, state, false);
+        return expanded_node;
+    }
+
+    std::shared_ptr<SearchNode> simpleChild(std::shared_ptr<SearchNode> node) {
+        auto firstChild = node->child_nodes.at(0);
+        return firstChild;
+    }
+};
+
+TEST_F(UCT_TreePolicyTest, ExpandNode) {
+    // Arrange
+    State rootChild3 = State("rootChild3");
+    std::vector<State> unvisited_child_states{rootChild3};
+    root->set_unvisited_child_states(unvisited_child_states);
+
+    // Act
+    auto expandedNode = treePolicy.treePolicy(root);
+
+    // Assert
+    EXPECT_EQ(expandedNode->parent, root.get());
+    EXPECT_EQ(root->child_nodes.size(), 3);
```

Comment:
I don't understand why root has 3 child nodes?

Reply:
Because it had 2 and then the tests expanded a new state to become the third child node